flow/project.py

```diff
@@ -2541,9 +2790,22 @@ def run(self, jobs=None, names=None, pretend=False, np=None, timeout=None, num=N
         :type ignore_conditions:
             :py:class:`~.IgnoreConditions`
         """
-        # If no jobs argument is provided, we run operations for all jobs.
-        if jobs is None:
-            jobs = self
+        if jobs is not None:
+            aggregates = set()
+            for job in jobs:
+                if isinstance(job, signac.contrib.job.Job):
+                    aggregates.add((job,))
+                else:
+                    try:
+                        aggregates.add(tuple(job))  # An aggregate provided by the user
+                    except Exception:
+                        raise ValueError('Invalid aggregate provided by the user. If you want to '
+                                         'provide an aggregate, please pass a tuple of jobs '
+                                         'instead.')
```

Comment:
What kind of error are we trying to catch here? This seems overly defensive with the broad except.

Reply:
I'd say the most appropriate catch would be a `TypeError` or a `ValueError`.
I will add these